lib/actions/wallet.js

```diff
@@ -86,37 +86,26 @@ function setSendTXFailure(walletId, accountId, options) {
 
 /**
  * Get wallets using wallet client
- * @param {string} - type
  * @dispatches
  * @returns {void}
  */
-function getWallets(type = 'standard') {
-  assert(SUPPORTED_WALLET_TYPES.includes(type));
+function getWallets() {
   return async dispatch => {
-    const client = getClient();
+    const { wallet, multisig } = getClient();
 
-    let wallets = [];
-    let multisigWallets = [];
-
-    switch (type) {
-      case 'standard':
-        wallets.push(...(await client.wallet.getWallets()));
-        break;
-      case 'multisig':
-        multisigWallets.push(...(await client.multisig.getWallets()));
-        break;
-      case 'all':
-        wallets.push(...(await client.wallet.getWallets()));
-        multisigWallets.push(...(await client.multisig.getWallets()));
-        break;
-      default:
-        throw new Error(`invalid argument: ${type}`);
-    }
+    // wallet.getWallets returns all wallets, including multisig
+    // multisig.getWallets returns only multisig wallets
+    const [wallets, multisigs] = await Promise.all([
+      wallet.getWallets(),
+      multisig.getWallets(),
+    ]);
+
+    // filter to only standard wallets
+    const standard = wallets.filter(w => !multisigs.includes(w));
```

Comment:
It looks like set operations iterate over everything as well: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
I'm not quite sure how to get around this, we could use webworkers if we need to, have been thinking of spawning a global worker pool and then sending tasks to it and having it return promises so that its an easy api for doing multithreaded stuff

Reply:
For the multisig, you should add checks on state.clients.currentClient.services. The issue was fixed where an explicitly disabled service wasn't returning `false` so that should at least work until we have a better way to check if a multisig server is accessible. 

I think that's a great idea for a feature to consider for the platform- having a global worker pool that plugins can hook into for more intensive operations. That said, is there a benefit to having a global one rather than plugins spawning their own as necessary? 

As for the performance of this filter, there might be a couple possible solutions. A roundabout way that might help a little bit (and could be overkill for something so simple) would be to have the array sorted then you could do something like a binary search. The best thing though would be if these were in an object, and then the check would be constant time. I also wonder if a selector might be useful here (especially with the benefits of memoization). Having a selector at least for the `standard` wallets could be good, so at least if this operation needs to be done as is, it can be cached and retrieved as needed.

Reply:
currentclient.services

Reply:
The idea behind having a global one would be so that plugin devs wouldn't have to implement their own. The pool could be started at runtime and you could configure how many workers you want at runtime, then any plugin can just submit tasks and get a promise back and the pool will manage spawning workers and such. Trying to think of how to implement things only once such that everybody will be able to benefit from them

Reply:
I think in practice at small scale the current code would be faster than sorting and searching both lists. Is it ok for trying to optimize this in a polish sprint?

Reply:
Yeah, I think that's fine. Maybe we should start a catch-all for polish priorities. I do think having `standard` as a selector (i.e. memoized via `reselect`) though could be useful for use throughout this plugin.

Reply:
and yeah, +100 on the global worker pool idea.